src/wallet-account.ts

```diff
@@ -307,7 +307,7 @@ export class ConnectedWalletAccount extends Account {
             try {
                 return await super.signAndSendTransaction({ receiverId, actions });
             } catch (e) {
-                if (e.type === 'NotEnoughBalance') {
+                if (e.type === 'NotEnoughAllowance') {
```

Comment:
@mehtaphysical do you think we should add a simple explanation of what is happening in this piece of code? It's not obvious why we are checking for `NotEnoughAllowance` or `NotEnoughBallance` here.

Reply:
I'm down with adding comments, too. And thanks for jumping on this issue